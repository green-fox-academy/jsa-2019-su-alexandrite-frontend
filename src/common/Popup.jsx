import React from 'react';
import { Ionicons } from '@expo/vector-icons';
import {
  Text,
  View,
  Modal,
  SafeAreaView,
  TouchableHighlight,
} from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';

const PopupHeader = ({ title, onCancel }) => (
  <View style={styles.modalHeader}>
    <Text style={styles.modalTitle}>{title}</Text>
    <TouchableHighlight style={styles.modalCloseButtonWrapper} underlayColor="rgba(0,0,0,0.1)" onPress={onCancel}>
      <Ionicons name="md-close" size={13} color="white" />
    </TouchableHighlight>
  </View>
);

const PopupButtonGroup = ({
  onConfirm, onCancel, confirmButtonText, confirmDisabled,
}) => (
  <View style={styles.modalButtonGroup}>
    <TouchableHighlight
      style={[
        styles.modalButton,
        { backgroundColor: '#5eceb1' },
        confirmDisabled && { opacity: 0.5 },
      ]}
      disabled={confirmDisabled}
      onPress={onConfirm}
      underlayColor="#5eceb1a0"
    >
      <Text style={{ fontSize: 14, color: '#fff' }}>{confirmButtonText}</Text>
    </TouchableHighlight>
    <TouchableHighlight
      style={styles.modalButton}
      onPress={onCancel}
      underlayColor="#eee"
    >
      <Text style={{ color: '#666' }}>Cancel</Text>
    </TouchableHighlight>
  </View>
);

export default function Popup({
  visible, onCancel, onConfirm, confirmButtonText, children, confirmDisabled, title,
}) {
  return (
    <Modal
      animationType="fade"
      transparent
      visible={visible}
      presentationStyle="overFullScreen"
      onRequestClose={onCancel}
    >
      <SafeAreaView style={styles.modalBackground}>
        <View style={styles.modalContainer}>
          <PopupHeader title={title} onCancel={onCancel} />
          <View style={styles.modalBody}>
            {children}
            <PopupButtonGroup
              confirmButtonText={confirmButtonText}
              confirmDisabled={confirmDisabled}
              onConfirm={onConfirm}
              onCancel={onCancel}
            />
          </View>
        </View>
      </SafeAreaView>
    </Modal>
  );
}
Popup.propTypes = {
  visible: PropTypes.bool.isRequired,
  onCancel: PropTypes.func.isRequired,
  onConfirm: PropTypes.func.isRequired,
  confirmButtonText: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
  confirmDisabled: PropTypes.bool,
  title: PropTypes.string,
};

Popup.defaultProps = {
  confirmDisabled: false,
  title: '',
};

PopupHeader.propTypes = {
  title: PropTypes.string.isRequired,
  onCancel: PropTypes.func.isRequired,
};

PopupButtonGroup.propTypes = {
  onCancel: PropTypes.func.isRequired,
  onConfirm: PropTypes.func.isRequired,
  confirmDisabled: PropTypes.bool.isRequired,
  confirmButtonText: PropTypes.string.isRequired,
};
