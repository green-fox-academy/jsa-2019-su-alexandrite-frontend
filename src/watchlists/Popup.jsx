import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { Ionicons } from '@expo/vector-icons';
import {
  Text,
  View,
  Modal,
  Button,
  TextInput,
  SafeAreaView,
  TouchableHighlight,
} from 'react-native';
import PropTypes from 'prop-types';
import { postWatchList } from '../redux/watchList/actionCreator';
import styles from './styles';

export default function Popup({ visible, toggle }) {
  const [watchListTitle, setWatchListTitle] = useState('');
  const dispatch = useDispatch();

  return (
    <Modal
      animationType="fade"
      transparent
      visible={visible}
      presentationStyle="overFullScreen"
      onRequestClose={() => {
        toggle(false);
      }}
    >
      <SafeAreaView style={styles.modalBackground}>
        <View style={styles.modalContainer}>
          <View style={styles.modalHeader}>
            <Text style={styles.modalTitle}>Create a watchlist</Text>
            <TouchableHighlight style={styles.modalCloseButtonWrapper} underlayColor="rgba(0,0,0,0.1)" onPress={() => toggle(false)}>
              <Ionicons name="md-close" size={13} color="white" />
            </TouchableHighlight>
          </View>
          <View style={styles.modalBody}>
            <TextInput style={styles.modalTextField} placeholder="Watchlist Name" onChangeText={(text) => setWatchListTitle(text)} />
            <View style={styles.modalButtonGroup}>
              <Button
                style={styles.modalButton}
                title="Cancel"
                onPress={() => {
                  toggle(false);
                }}
              />
              <Button
                title="Add"
                style={styles.modalButton}
                disabled={watchListTitle === ''}
                onPress={() => {
                  toggle(!visible);
                  // setWatchListItem('');
                  if (watchListTitle === '') {
                    toggle(true);
                  } else {
                    dispatch(postWatchList(watchListTitle));
                  }
                }}
              />
            </View>
          </View>
        </View>
      </SafeAreaView>
    </Modal>
  );
}
Popup.propTypes = {
  visible: PropTypes.bool.isRequired,
  toggle: PropTypes.func.isRequired,
};
