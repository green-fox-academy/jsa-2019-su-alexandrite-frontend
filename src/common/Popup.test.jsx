import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

import Popup from './Popup';

describe('<Popup />', () => {
  it('should render properly and match the snapshot', () => {
    const tree = renderer.create(
      <Popup
        visible={false}
        onCancel={jest.fn()}
        onConfirm={jest.fn()}
        confirmButtonText="OK"
        confirmDisabled={false}
        title="Title"
      >
        <Text>Test</Text>
      </Popup>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render properly and match the snapshot when confirm enabled', () => {
    const tree = renderer.create(
      <Popup
        visible={false}
        onCancel={jest.fn()}
        onConfirm={jest.fn()}
        confirmButtonText="OK"
        confirmDisabled
        title="Title"
      >
        <Text>Test</Text>
      </Popup>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
