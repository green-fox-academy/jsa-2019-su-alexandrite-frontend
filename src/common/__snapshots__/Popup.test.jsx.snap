// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popup /> should render properly and match the snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  presentationStyle="overFullScreen"
  transparent={true}
  visible={false}
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,.3)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 6,
          "height": 280,
          "width": 285,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#4d63be",
            "borderTopRightRadius": 6,
            "borderTopStartRadius": 6,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 13,
              "fontWeight": "bold",
              "padding": 15,
            }
          }
        >
          Title
        </Text>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderTopRightRadius": 6,
              "padding": 15,
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
            "padding": 15,
          }
        }
      >
        <Text>
          Test
        </Text>
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 2,
                  "height": 40,
                  "justifyContent": "center",
                  "width": 255,
                },
                Object {
                  "backgroundColor": "#5eceb1",
                },
                false,
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 14,
                }
              }
            >
              OK
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 2,
                "height": 40,
                "justifyContent": "center",
                "width": 255,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#666",
                }
              }
            >
              Cancel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</Modal>
`;

exports[`<Popup /> should render properly and match the snapshot when confirm enabled 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  presentationStyle="overFullScreen"
  transparent={true}
  visible={false}
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,.3)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 6,
          "height": 280,
          "width": 285,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#4d63be",
            "borderTopRightRadius": 6,
            "borderTopStartRadius": 6,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 13,
              "fontWeight": "bold",
              "padding": 15,
            }
          }
        >
          Title
        </Text>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderTopRightRadius": 6,
              "padding": 15,
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
            "padding": 15,
          }
        }
      >
        <Text>
          Test
        </Text>
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 2,
                  "height": 40,
                  "justifyContent": "center",
                  "width": 255,
                },
                Object {
                  "backgroundColor": "#5eceb1",
                },
                Object {
                  "opacity": 0.5,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 14,
                }
              }
            >
              OK
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 2,
                "height": 40,
                "justifyContent": "center",
                "width": 255,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#666",
                }
              }
            >
              Cancel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</Modal>
`;
